### PR TITLE
Azure: list of clouds in frontend AzureSettings

### PR DESCRIPTION
**Pull request by kostrse**
This PR extends `@grafana/runtime`'s `AzureSettings` to include an optional list of clouds which could be sent to frontend when its different from the predefined list of clouds.

Related PR on backend: #83460

Will be used by the Azure frontend SDK: https://github.com/grafana/grafana-azure-sdk-react/pull/128

### CODE REVIEW

**Review by aangelisc — APPROVED** [review]
👍 from me pending review from a codeowner.

File: packages/grafana-runtime/src/config.ts
Comment (sunker) [inline packages/grafana-runtime/src/config.ts:23]: Should this prop be marked as deprecated?

**Review by sunker — APPROVED** [review]
Looks good to me. Just a question.

File: packages/grafana-runtime/src/config.ts
Comment (kostrse) [reply packages/grafana-runtime/src/config.ts:23]: They both are needed. `cloud` is the default cloud, and `clouds` is a list of available clouds.
`cloud` should have been named `defaultCloud` and this is what it is.

**Pull request merged**
